konf.py

```diff
@@ -0,0 +1,139 @@
+#! /usr/bin/env python3
+
+import yaml
+import jinja2
+import argparse
+
+
+# Custom Jinja2 functions
+@jinja2.contextfunction
+def getSiteName(context, addStaginPrefix=False):
+    if context['data'].get('name'):
+        name = context['data'].get('name')
+    else:
+        domain = context['domain'].split(".")
+
+        if addStaginPrefix and context["deploymentEnv"] == "staging":
+            domain.insert(-2, 'staging')
+
+        name = "-".join(domain)
+
+    return name
+
+
+@jinja2.contextfilter
+def siteDomain(context, s):
+    domain = s.split(".")
+
+    if context["deploymentEnv"] == "staging":
+        domain.insert(-2, 'staging')
+
+    return ".".join(domain)
+
+
+@jinja2.contextfilter
+def isApexDomain(context, s):
+    return s.count('.') == 1
+
+
+class Konf:
+    def __init__(
+        self,
+        values_file,
+        env,
+        local_qa=False,
+        docker_tag=None
+    ):
+        self.deploymentEnv = env
+
+        # Init Jinja2 environment
+        templateLoader = jinja2.FileSystemLoader("./templates")
+        self.tplEnv = jinja2.Environment(loader=templateLoader)
+
+        # Add custom jinja2 functions and filters
```

Comment:
As far as I can tell you only call `self.tplEnv.render` in one place. Did you add these functions here in the global space for performance or something? Because they could be passed at render time couldn't they?